src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -26,28 +29,36 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public partial class Solution
     {
-        // SolutionState that doesn't hold onto Project/Document
-        private readonly SolutionState _state;
+        private readonly SolutionCompilationState _compilationState;
 
         // Values for all these are created on demand.
         private ImmutableHashMap<ProjectId, Project> _projectIdToProjectMap;
 
-        private Solution(SolutionState state)
+        private Solution(SolutionCompilationState state)
         {
             _projectIdToProjectMap = ImmutableHashMap<ProjectId, Project>.Empty;
-            _state = state;
+            _compilationState = state;
         }
 
-        internal Solution(Workspace workspace, SolutionInfo.SolutionAttributes solutionAttributes, SolutionOptionSet options, IReadOnlyList<AnalyzerReference> analyzerReferences)
-            : this(new SolutionState(workspace.Kind, workspace.PartialSemanticsEnabled, workspace.Services.SolutionServices, solutionAttributes, options, analyzerReferences))
+        internal Solution(
+            Workspace workspace,
+            SolutionInfo.SolutionAttributes solutionAttributes,
+            SolutionOptionSet options,
+            IReadOnlyList<AnalyzerReference> analyzerReferences)
+            : this(new SolutionCompilationState(
+                new SolutionState(workspace.Kind, workspace.Services.SolutionServices, solutionAttributes, options, analyzerReferences),
+                workspace.PartialSemanticsEnabled))
         {
         }
 
-        internal SolutionState State => _state;
+        private SolutionState _state => this.State;
```

Comment:
ugly bridge property so i don't have to cahnge a ton of lines in this file.  a followup PR/commit will remove this and update all usages of `_state` to just be `this.State`.